version: '3'

services:

  db:
    image: postgres
    container_name: postgres_db
    restart: always
    environment:
      PGDATA: /var/lib/postgresql/data/pgdata
    env_file:
      - .env
    ports:
      - '5435:5432'                                     # порты: 5435 - внешний, 5432 - внутренний
    healthcheck:
      test: [ 'CMD-SHELL', 'pg_isready -U postgres' ]
      interval: 10s
      timeout: 5s
      retries: 5
    volumes:
      - ./pg_data:/var/lib/postgresql/data/pgdata

  redis:
    image: redis
    restart: on-failure
    ports:
      - '6379:6379'

  app:
    build: .
    tty: true
    command: sh -c 'python manage.py migrate && python manage.py runserver 0.0.0.0:8000'
    ports:
      - '8001:8000'
    env_file:
      - .env
    volumes:
      - .:/code
    depends_on:
      db:
        condition: service_healthy

  celery:
    build: .
    tty: true
    command: celery -A config worker -l INFO
    restart: on-failure
    depends_on:
      - redis
      - app

  celery-beat:
    build: .
    tty: true
    command: celery -A config.celery beat -l INFO -S django
    restart: on-failure
    depends_on:
      - db
      - redis
      - app


# Для БД volumes - pg_data
volumes:
  pg_data:
